zhaquirks/sonoff/zbm5.py

```diff
@@ -0,0 +1,166 @@
+            is_manufacturer_specific=True,
+        )
+        relay_3_detached = ZCLAttributeDef(
+            id=0x0FFC,
+            type=t.Bool,
+            is_manufacturer_specific=True,
+        )
+
+    def _update_attribute(self, attrid, value):
+        """Update attribute and handle relay mask conversion."""
+        super()._update_attribute(attrid, value)
+
+        if attrid == self.AttributeDefs.detach_relay_mask.id:
+            # Convert bitmap to individual relay states
+            mask = value
+            self._update_attribute(
+                self.AttributeDefs.relay_1_detached.id,
+                bool(mask & SonoffDetachedRelayMask.Relay1),
+            )
+            self._update_attribute(
+                self.AttributeDefs.relay_2_detached.id,
+                bool(mask & SonoffDetachedRelayMask.Relay2),
+            )
+            self._update_attribute(
+                self.AttributeDefs.relay_3_detached.id,
+                bool(mask & SonoffDetachedRelayMask.Relay3),
+            )
+
+    async def write_attributes(self, attributes, manufacturer=None, **kwargs):
+        """Handle writing individual relay attributes by updating the mask."""
+        # Check if any individual relay attributes are being written
+        mask_attr = self.AttributeDefs.detach_relay_mask.id
+        mask = self.get(mask_attr, 0)
+        new_attributes = attributes.copy()
+
+        relay_attr_defs = [
+            (self.AttributeDefs.relay_1_detached, SonoffDetachedRelayMask.Relay1),
+            (self.AttributeDefs.relay_2_detached, SonoffDetachedRelayMask.Relay2),
+            (self.AttributeDefs.relay_3_detached, SonoffDetachedRelayMask.Relay3),
+        ]
+        for attrid, value in attributes.items():
+            for attr_def, bit_mask in relay_attr_defs:
+                if attrid in (attr_def.id, attr_def.name):
+                    new_attributes.pop(attrid)
+                    if value:
+                        mask |= bit_mask
+                    else:
+                        mask &= ~bit_mask
+                    new_attributes[mask_attr] = mask
+                    break
+
+        return await super().write_attributes(new_attributes, manufacturer, **kwargs)
+
+
+# Base quirk for 1-channel device
+zbm_1c_quirk = (
+    QuirkBuilder("SONOFF", "ZBM5-1C-80/86")
+    .applies_to("SONOFF", "ZBM5-1C-120")
+    .adds(SonoffCluster)
+    .enum(
+        SonoffCluster.AttributeDefs.work_mode.name,
+        SonoffWorkMode,
+        SonoffCluster.cluster_id,
+        translation_key="work_mode",
+        fallback_name="Work mode",
+        entity_type=EntityType.DIAGNOSTIC,
+        entity_platform=EntityPlatform.SENSOR,
+        initially_disabled=True,
+    )
+    .switch(
+        SonoffCluster.AttributeDefs.relay_1_detached.name,
+        SonoffCluster.cluster_id,
+        translation_key="relay_1_detached",
+        fallback_name="Detach Relay 1",
+        entity_type=EntityType.CONFIG,
+        initially_disabled=True,
```

Comment:
Same here. We'll also need to lower-case "relay" to confirm with HA guidelines, as it's mid-sentence.
(The fallback name is the name we'll also use to add to Home Assistant's `strings.json` for translations.)

I'd also suggest changing the translation_key to be more in line with the name and existing translation keys. 
```suggestion
        entity_type=EntityType.CONFIG,
        initially_disabled=True,
        translation_key="detach_relay_1",
        fallback_name="Detach relay 1",
```
We could also use translation placeholders instead.

These entities should already be in the `CONFIG` category by default, since that's the default for quirks v2 switches. v2 quirks are mostly made for config entities.